nistats/first_level_model.py

```diff
@@ -441,22 +448,27 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
 
         return self
 
-    def compute_contrast(self, contrast_def, contrast_name=None,
-                         stat_type=None, output_type='z_score'):
+    def compute_contrast(self, contrast_def, stat_type=None,
+                         output_type='z_score'):
         """Generate different outputs corresponding to
         the contrasts provided e.g. z_map, t_map, effects and variance.
         In multi-session case, outputs the fixed effects map.
 
         Parameters
         ----------
-        contrast_def : array or list of arrays of shape (n_col) or (n_run, n_col)
+        contrast_def : str or array of shape (n_col) or list of (string or
+                       array of shape (n_col))
             where ``n_col`` is the number of columns of the design matrix,
             (one array per run). If only one array is provided when there
             are several runs, it will be assumed that the same contrast is
-            desired for all runs
-
-        contrast_name : str, optional
-            name of the contrast
+            desired for all runs. The string can be a formula compatible with
+            the linear constraint of the Patsy library. Basically one can use
+            the name of the conditions as they appear in the design matrix of
+            the fitted model combined with operators /*+- and numbers.
+            Nonetheless all contrasts assume the baseline of the tested effect
+            as 0, so specifying a different baseline in a formula will have no
+            effect. Please checks the patsy documentation for formula examples:
```

Comment:
I am not sure that the baseline question should be discussed here (looks like a corner case).